bentoml/yatai/deployment/aws_ec2/operator.py

```diff
@@ -0,0 +1,785 @@
+                SecurityGroupIds:
+                - !GetAtt SecurityGroupResource.GroupId
+
+    TargetGroup:
+        Type: AWS::ElasticLoadBalancingV2::TargetGroup
+        Properties:
+            VpcId: !Ref Vpc1
+            Protocol: HTTP
+            Port: 5000
+            TargetType: instance
+            HealthCheckEnabled: true
+            HealthCheckIntervalSeconds: {target_health_check_interval_seconds}
+            HealthCheckPath: {target_health_check_path}
+            HealthCheckPort: {target_health_check_port}
+            HealthCheckProtocol: HTTP
+            HealthCheckTimeoutSeconds: {target_health_check_timeout_seconds}
+            HealthyThresholdCount: {target_health_check_threshold_count}
+
+    LoadBalancerSecurityGroup:
+        Type: AWS::EC2::SecurityGroup
+        Properties:
+            GroupDescription: "security group for loadbalancing"
+            VpcId: !Ref Vpc1
+            SecurityGroupIngress:
+                -
+                    IpProtocol: tcp
+                    CidrIp: 0.0.0.0/0
+                    FromPort: 80
+                    ToPort: 80
+
+    InternetGateway:
+        Type: AWS::EC2::InternetGateway
+
+    Gateway:
+        Type: AWS::EC2::VPCGatewayAttachment
+        Properties:
+            InternetGatewayId: !Ref InternetGateway
+            VpcId: !Ref Vpc1
+
+    PublicRouteTable:
+        Type: AWS::EC2::RouteTable
+        Properties:
+            VpcId: !Ref Vpc1
+
+    PublicRoute:
+        Type: AWS::EC2::Route
+        DependsOn: Gateway
+        Properties:
+            DestinationCidrBlock: 0.0.0.0/0
+            GatewayId: !Ref InternetGateway
+            RouteTableId: !Ref PublicRouteTable
+
+    RouteTableSubnetTwoAssociationOne:
+        Type: AWS::EC2::SubnetRouteTableAssociation
+        Properties:
+          RouteTableId: !Ref PublicRouteTable
+          SubnetId: !Ref Subnet1
+    RouteTableSubnetTwoAssociationTwo:
+        Type: AWS::EC2::SubnetRouteTableAssociation
+        Properties:
+          RouteTableId: !Ref PublicRouteTable
+          SubnetId: !Ref Subnet2
+
+    Vpc1:
+        Type: AWS::EC2::VPC
+        Properties:
+            CidrBlock: 172.31.0.0/16
+            EnableDnsHostnames: true
+            EnableDnsSupport: true
+            InstanceTenancy: default
+
+    Subnet1:
+        Type: AWS::EC2::Subnet
+        Properties:
+            VpcId: !Ref Vpc1
+            AvailabilityZone:
+                Fn::Select:
+                    - 0
+                    - Fn::GetAZs: ""
+            CidrBlock: 172.31.16.0/20
+            MapPublicIpOnLaunch: true
+
+    Subnet2:
+        Type: AWS::EC2::Subnet
+        Properties:
+            VpcId: !Ref Vpc1
+            AvailabilityZone:
+                Fn::Select:
+                    - 1
+                    - Fn::GetAZs: ""
+            CidrBlock: 172.31.0.0/20
+            MapPublicIpOnLaunch: true
+
+    LoadBalancer:
+        Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+        Properties:
```

Comment:
Did you consider using the bentoml deployment namespace + deployment name as the NAME of this LoadBalancer resource on AWS?  That way the ELB URL will be more a lot more recognizable for the end-user. IIUC, currently the ELB endpoint URL is an auto-generated random hash.

Related docs about loadbalancer name:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-loadbalancer.html
```
The name of the load balancer. This name must be unique per region per account, can have a maximum of 32 characters, must contain only alphanumeric characters or hyphens, must not begin or end with a hyphen, and must not begin with "internal-".

If you don't specify a name, AWS CloudFormation generates a unique physical ID for the load balancer. If you specify a name, you cannot perform updates that require replacement of this resource, but you can perform other updates. To replace the resource, specify a new name.

Required: No

Type: String

Update requires: Replacement
```

Reply:
I always advice against giving custom constant names to resources because updating stack fails which modifies those resources.
for example if we wish to add more different target groups for spots, update will fail, whereas without it will be just replaced with new one.
What do you think?

Reply:
Could you explain what do you mean by "updating stack fails which modifies those resources."?

I was thinking of only adding NAME to the ELB resource because it affects the endpoint URL that the user will interact with. As you pointed out in the previous comment, ELB stays the same when doing updates:

> No it wont change,because only scaling-group replace instances,balancer stays same.
This will only change if balancer is replaced which doesn't happen in deployment cycle of bentos.

The reason I'm asking for this is that for users who have multiple deployments, having the deployment name in the endpoint URL is just easier to deal with.

Reply:
Yeah,by current cycle is deploy,update,delete which does not change ELB,but suppose we add more features to it and users already have stacks deployed they wont be able to just update it with added new features if we set constant name to it.
If we set random name,it will replace ELB even if only bento version is changed in update.

Reply:
How about adding the random chars only the first time the ELB is created?  Isn't it the current and default behavior to use a randomly generated name in ELB? I think the desired behavior on BentoML side is just adding the deployment name to be part of the ELB NAME, so at the end of the day, the deployment name string is part of the endpoint URL

Reply:
So it's basically setting constant name?

Reply:
added name.